pkg/cloudprovider/aws.go

```diff
@@ -378,3 +385,53 @@ func getInstanceIdFromProviderId(providerId string) (string, error) {
 
 	return instanceId, nil
 }
+
+// sharedCredentialsFileFromDirectory returns a location (path) to the shared credentials file that was created
+// using the data in the provided directory.
+// This is heavily inspired and partially copied verbatim from openshift/cluster-ingress-operator, file
+// pkg/util/aws/shared_credentials_file.go.
+// If file ${dir}/credentials exists then sharedCredentialsFileFromDirectory simply returns that string.
+// Otherwise if such a file does not exist, sharedCredentialsFileFromDirectory will attempt to build a credentials
+// file from ${dir}/aws_access_key_id and ${dir}/aws_secret_access_key.
+func sharedCredentialsFileFromDirectory(dir string) (string, error) {
+	fCredentials := filepath.Join(dir, "credentials")
+	fAccessKeyID := filepath.Join(dir, "aws_access_key_id")
+	fSecretAccessKey := filepath.Join(dir, "aws_secret_access_key")
+
+	if _, err := os.Stat(fCredentials); err == nil {
+		return fCredentials, nil
+	}
+
+	klog.Infof("Could not find file %s, trying to build temp file from %s and %s",
+		fCredentials, fAccessKeyID, fSecretAccessKey)
+
+	accessKeyID, err := os.ReadFile(fAccessKeyID)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to read file %s", fAccessKeyID)
+	}
+	secretAccessKey, err := os.ReadFile(fSecretAccessKey)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to read file %s", fSecretAccessKey)
+	}
+	data := newConfigForStaticCreds(string(accessKeyID), string(secretAccessKey))
+
+	f, err := ioutil.TempFile("", "aws-shared-credentials")
```

Comment:
ioutil.TempFile is deprecated https://pkg.go.dev/io/ioutil#TempFile, can we use `os.CreateTemp` instead?

Reply:
thanks done